### PR TITLE
VILA Added

### PR DESCRIPTION
**Pull request by gaurangbharti1**

### CODE REVIEW

File: main_pipeline.py [hunk 1]
```diff
@@ -28,33 +28,36 @@ def get_video_info(video_path):
     duration = frame_count / fps
     return vr, duration, fps
 
-def calculate_keyframes(video_duration, frame_rate, start_time, end_time):
+def calculate_keyframes(video_duration, frame_rate, start_time, end_time, visual_detail="high"):
```
Comment (mvoodarla) [inline main_pipeline.py:31]: can we restructure this function. super confused by all these constants and random if statements. make it more understandable.

File: main_pipeline.py [hunk 2]
```diff
@@ -87,7 +90,7 @@ def main(
     """
     :param video: The video to be described
     :param conciseness: The level of detail for the final description. Pick from 'concise', 'medium', or 'detailed'
-    :param visual_detail: The level of visual detail for the final description. Pick from 'low', 'medium', or 'high'
+    :param visual_detail: The level of visual detail for the final description. Pick from 'low', 'medium', 'high', or 'ultra'
```
Comment (mvoodarla) [inline main_pipeline.py:93]: add a period at the end

File: main_pipeline.py [hunk 3]
```diff
@@ -214,14 +226,26 @@ def process_keyframe(keyframe_path):
         keyframe_job = model.push(file_arg, detail_prompt)
         return keyframe_job
 
+    # only used if VILA is chosen
```
Comment (mvoodarla) [inline main_pipeline.py:229]: what is this doing exactly, a bit confusing to me.

File: main_pipeline.py [hunk 4]
```diff
@@ -214,14 +226,26 @@ def process_keyframe(keyframe_path):
...
+    def process_scenes(scene_frames):
+        model = model_mapping[visual_detail]
+        file_arg = file_type[visual_detail](path=video_path)
+        keyframe_numbers = str(scene_frames)
+        keyframe_numbers = keyframe_numbers.replace("[", "").replace("]", "").replace(" ", "")
```
Comment (mvoodarla) [inline main_pipeline.py:234]: this line seems like a hack. why are you doing string manipulation to get numbers?

File: main_pipeline.py (hunk 1 above)
Comment (gaurangbharti1) [reply main_pipeline.py:31]: there was only one more if statement added and some more constants that just represent the locations of the frames. will write them separately so it's easier to read

File: main_pipeline.py (hunk 4 above)
Comment (gaurangbharti1) [reply main_pipeline.py:234]: the input to VILA rn is a string that has the frame numbers in the form of comma-separated values. So, once we perform all the processing as lists, this converts it to a format that can be processed by VILA directly. I can make it cleaner though, will do that, but that step of list -> string is necessary to pass it into vila correctly.

**Pull request merged**